crypto/encoding/codec.go

```diff
@@ -57,9 +59,9 @@ func PubKeyToProto(k crypto.PubKey) (pc.PublicKey, error) {
 				Secp256K1: k,
 			},
 		}
-	case *bls12381.PubKey:
+	case *bls12381.PubKey, bls12381.PubKey:
```

Comment:
Do we need to keep the `*bls12381.PubKey`?
The declaration in the method is clearly not a pointer

Reply:
The declaration in the method is an interface.

Reply:
Yeah, so if it's an interface, does it make sense to leave `*bls12381.PubKey` (a pointer to an interface)?

Reply:
`*bls12381.PubKey` is a pointer to a struct. I would leave both (a pointer to a struct and struct itself) to be safe

Reply:
OK, fair enough. Approving

Reply:
I couldn't find a case where a concrete type `bls12381.PubKey` would make into this method based on the current implementation.

Another approach would be to have used the same as in `PubKeyFromTypeAndBytes` leveraging the `k.Type()` something like

```
	switch k.Type() {
	case bls12381.KeyType:
		//...
	} 
```

Reply:
> Another approach would be to have used the same as in PubKeyFromTypeAndBytes leveraging the k.Type() something like

Great idea!